src/l500/l500-color.cpp

```diff
@@ -485,6 +499,62 @@ namespace librealsense
     }
 
 
+    void l500_color_sensor::set_default_controls_for_calibration_if_needed()
+    {
+        for (auto && calib_control : _calib_controls)
+        {
+            auto && control = get_option(calib_control.option);
+
+            auto curr_val = control.query();
+            if (curr_val != calib_control.default_value)
+            {
+                LOG_INFO( "Calibration changed option: " << rs2_option_to_string( calib_control.option )
+                                              << " from: " << curr_val
+                                              << " to: " << calib_control.default_value );
+                calib_control.need_to_restore = true;
+                calib_control.previous_value = curr_val;
+                control.set(calib_control.default_value);
+            }
+        }
+    }
+
+    void l500_color_sensor::restore_pre_calibration_controls_if_needed()
+    {
+        for (auto && calib_control : _calib_controls)
+        {
+            auto && control = get_option(calib_control.option);
+
+            auto curr_val = control.query();
+            if( calib_control.need_to_restore &&
```

Comment:
Can we get a log entry ever for options that do not need restoration? I'd like the logic to be verbose

Reply:
Yes - I will add it

Reply:
Done